src/main/scala/com/itechart/ny_accidents/service/metric/DayPeriodMetricService.scala

```diff
@@ -0,0 +1,52 @@
+package com.itechart.ny_accidents.service.metric
+
+import java.util.Calendar
+
+import com.itechart.ny_accidents.constants.GeneralConstants._
+import com.itechart.ny_accidents.entity.MergedData
+import com.itechart.ny_accidents.service.DayPeriodService
+import org.apache.spark.rdd.RDD
+
+object DayPeriodMetricService {
+  private val sunriseSchedule = DayPeriodService.sunrisesSunsets
+
+  def getFrequency(data: RDD[MergedData]): Map[String,Long] ={
+    val filteredData = data.collect()
+      .filter(_.accident.dateTime.isDefined)
+      .map(_.accident.dateTime.get).groupBy(date => (date.getDate, date.getMonth)).map(t => (countMorningAndEveningSeconds(t._2.head), checkAccidents(t._2.toSeq)))
+    val preparedData = filteredData.reduce((l, r) =>
+      ((l._1._1 + r._1._1, l._1._2 + r._1._2), (l._2._1 + r._2._1, l._2._2 +r._2._2 )))
```

Comment:
This is unreadable and needs to be refactored

Reply:
done using pattern matching
https://github.com/itechartbigdatalab/NY-Accidents-Rate/commit/8ae0ac58c5f32b5050339915a347df486b0dc776